core/src/main/java/me/mykindos/betterpvp/core/world/blocks/WorldBlockHandler.java

```diff
@@ -55,16 +63,22 @@ public void addRestoreBlock(@Nullable LivingEntity entity, Block block, Material
                 restoreBlock.setSummoner(entity);
             }
             restoreBlock.setExpire(force ? newExpiry : Math.max(restoreBlock.getExpire(), newExpiry));
+            if (restoreBlock.getNewMaterial() == Material.AIR) {
+                restoreBlock.setNewMaterial(newMaterial);
+                block.setType(newMaterial);
+            }
         } else {
-            RestoreBlock newRestoreBlock = new RestoreBlock(block, newMaterial, expiry, entity, label);
+            if (block.getType().equals(Material.WATER) && !newMaterial.equals(Material.WATER)) {
+                Block aboveBlock = block.getLocation().clone().add(0, 1, 0).getBlock();
+                if (aboveBlock.getType().equals(Material.LILY_PAD)) {
+                    addRestoreBlock(entity, aboveBlock, Material.AIR, expiry, force, label);
+                }
```

Comment:
Instead of this specific case, I think you can do this (from my research)

if(!aboveBlock.getType().isSolid()) {
   addRestoreBlock(entity, aboveBlock, Material.AIR, expiry, force, label);
}

Reply:
done